### DISCUSSION TITLE
Osu!Mania LN have very poor Performance

### DISCUSSION THREAD

**Discussion by retlom2** [ppy/osu#21968] [Q&A]
### Type

Performance

### Bug description

If you play any song with a bunch of LNs the fps drops from above 1k to 600-700 as soon as the LN part is over back to 1k+ FPS on my system.
This results in to visible Microstutters and lag/ghosting

CPU: AMD Ryzen 9 3900X 12-Core Processor 3.79 GHz
GPU RTX2060
RAM 32gb
Win10 Newest update

### Screenshots or videos

Before LN
![BeforLN](https://user-images.githubusercontent.com/73026884/210120299-33b99566-3af1-4865-91d1-3407827a0649.PNG)
During
![DuringLN](https://user-images.githubusercontent.com/73026884/210120300-7fb6cf00-e1c4-409a-a1a1-31b624cf7444.PNG)

Screenshots where taken a round 1:25 AM for the Logs there is nothing to see at that time :) 

### Version

2022.1228.0-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/10326643/database.log)
[input.log](https://github.com/ppy/osu/files/10326644/input.log)
[performance.log](https://github.com/ppy/osu/files/10326645/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10326646/runtime.log)
[updater.log](https://github.com/ppy/osu/files/10326647/updater.log)

**Comment by bdach** [comment]
Please link the beatmaps that you're seeing this on.

**Reply by ghost** [reply]
Not OP but looks like https://osu.ppy.sh/beatmapsets/489909#mania/1044377

**Reply by retlom2** [reply]
it dosnt really matter the Beatmap tho as soon as there are a lot of LN it starts to happn

**Comment by peppy** [comment]
There's definitely more overhead with hold notes which shows up on both update and draw threads. Will require further investigation (wondering if it's purely from more drawables being handled/invalidated by `ScrollingPlayfield` or something deeper)..
